Source/SwiftLintFramework/Rules/Lint/UnusedPrivateDeclarationRule.swift

```diff
@@ -98,6 +122,14 @@ private extension File {
             if cursorInfo["key.overrides"] != nil {
                 return nil
             }
+
+            // Skip CodingKeys as they are used
+            if kind == .enum,
+                cursorInfo.name == "CodingKeys",
+                let annotatedDecl = cursorInfo["key.annotated_decl"] as? String,
+                annotatedDecl.contains(">CodingKey<") {
```

Comment:
It was the "best" way to find if it did conform to a protocol.

Reply:
On my phone right now so not 100% sure but you should be able to use `key.inheritedtypes` or `key.inherited_types` for this. If you search the SwiftLint or SourceKitten codebases you’ll find usage examples.

Reply:
Ah so that won't work because we just have `cursorInfo` here. However, we can strengthen this check by making sure it's the right `CodingKey` protocol.

```swift
annotatedDecl.contains("usr=\"s:s9CodingKeyP\">CodingKey<") {
```